### DISCUSSION TITLE
Why not executable modules in `console_scripts`?

### DISCUSSION THREAD

**Discussion by abitrolly** [pypa/setuptools#2690] [Q&A] [answered]
Currently, console scripts that are defined as a call to a single file, produce an error when executed.
```python
      entry_points={
          'console_scripts': [
              'sqlite2png=sqlalchemy_schemadisplay',
           ],
      },
```

```python
✗ sqlite2png
Traceback (most recent call last):
  File "/home/anatoli/.local/share/virtualenvs/sqlite2png-67cYiUtQ/bin/sqlite2png", line 33, in <module>
    sys.exit(load_entry_point('sqlalchemy-schemadisplay', 'console_scripts', 'sqlite2png')())
TypeError: 'module' object is not callable
```

I can wrap my `if __name__ == '__main__'` logic into `main()` function and define the entrypoint with it.
```python
      entry_points={
          'console_scripts': [
              'sqlite2png=sqlalchemy_schemadisplay:main',
           ],
      },
```

But why not just make the whole module executable as a script?

**Comment by webknjaz** [comment, marked as answer]
I don't see much use for this. Honestly, I'd prefer a runpy-style invocation (as in `python -m importable.thing`)

Besides, not having the "if main" thing is really an antipattern — import-time logic is evil.
